packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1490,9 +1481,10 @@ void ArbCore::operator()() {
                     auto add_status = addMessages(message_data, cache);
                     if (!add_status.status.ok()) {
                         core_error_string = add_status.status.ToString();
-                        message_data_status = MESSAGES_ERROR;
+                        core_error = true;
                         std::cerr << "ArbCore addMessages error: "
                                   << core_error_string << "\n";
+                        break;
```

Comment:
The breaks make me a bit nervous since we could accidentally have them in a nested for loop and not exit. Might be worth switching over to `return` with a scope guard approach. Something like
```c++
    class CoreShutdown {
        ArbCore &core;
        CoreShutdown(ArbCore& core): core(core_) {}
        ~CoreShutdown() { core.abortThread(); }
    };
    CoreShutdown shutdown{*this};
```

One place where a golang `defer` would be simpler

Reply:
Will handle refactor in separate PR